tests/utils/test_thumbnail_video_utils.py

```diff
@@ -0,0 +1,679 @@
+                                height=height,
+                                width=width,
+                                valid=True,
+                                mask=[list(row) for row in mask]))
+
+    else:
+        other_roi = None
+
+    if isinstance(roi_color, str):
+        roi_color = dict()
+        roi_color[0] = (255, 0, 0)
+        if other_roi is not None:
+            rng = np.random.default_rng(111)
+            for roi in other_roi:
+                color = tuple(rng.integers(0, 255, size=3))
+                roi_color[roi['id']] = color
+
+    thumbnail = _thumbnail_video_from_ROI_array(
+                    example_video,
+                    roi,
+                    other_roi=other_roi,
+                    roi_color=roi_color,
+                    padding=padding,
+                    tmp_dir=pathlib.Path(tmpdir),
+                    quality=9,
+                    timesteps=timesteps)
+
+    origin, fov = video_bounds_from_ROI(roi,
+                                        example_video.shape[1:3],
+                                        padding)
+
+    assert thumbnail.origin == origin
+    assert thumbnail.frame_shape == fov
+
+    rowmin = thumbnail.origin[0]
+    rowmax = thumbnail.origin[0]+thumbnail.frame_shape[0]
+    colmin = thumbnail.origin[1]
+    colmax = thumbnail.origin[1]+thumbnail.frame_shape[1]
+    assert rowmin <= y0
+    assert rowmax >= y0+height
+    assert colmin <= x0
+    assert colmax >= x0+width
+
+    assert thumbnail.video_path.is_file()
+
+    read_data = imageio.mimread(thumbnail.video_path)
+    assert len(read_data) == n_t
+
+    # factor of 4 reflects the upscaling of video frame sizes
+    frame_shape = thumbnail.frame_shape
+    if frame_shape[0] < 128 or frame_shape[1] < 128:
+        assert read_data[0].shape == (2*thumbnail.frame_shape[0],
+                                      2*thumbnail.frame_shape[1],
+                                      3)
+    else:
+        assert read_data[0].shape == (thumbnail.frame_shape[0],
+                                      thumbnail.frame_shape[1],
+                                      3)
+
+
+@pytest.mark.parametrize("min_max, quantiles, timesteps",
+                         [((50, 650), None, None),
+                          ((50, 650), None, np.arange(22, 57)),
+                          ((50, 650), None, None),
+                          ((50, 650), None, np.arange(22, 57)),
+                          ((111, 556), None, None),
+                          ((111, 556), None, np.arange(22, 57)),
+                          (None, (0.1, 0.75), None),
+                          (None, (0.1, 0.75), np.arange(22, 57))
+                          ])
+def test_thumbnail_from_path(tmpdir,
+                             example_unnormalized_rgb_video,
+                             example_unnormalized_rgb_video_path,
+                             min_max,
+                             quantiles,
+                             timesteps):
+    """
```

Comment:
Okay since this one shows up here. I've noticed that a lot of the functions and tests don't have at least a short, one sentence description of what is being tested or created in the numerous convenience functions. Please add them.